crates/ruff_linter/src/rules/flake8_use_pathlib/rules/dotless_pathlib_with_suffix.rs

```diff
@@ -0,0 +1,110 @@
+use crate::checkers::ast::Checker;
+use ruff_diagnostics::{AlwaysFixableViolation, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, ViolationMetadata};
+use ruff_python_ast::{Expr, ExprAttribute, ExprCall, ExprStringLiteral, StringFlags};
+use ruff_python_semantic::analyze::typing;
+use ruff_python_semantic::SemanticModel;
+use ruff_text_size::Ranged;
+
+/// ## What it does
+/// Checks for `Path.with_suffix()` calls where
+/// the given suffix does not have a leading dot.
+///
+/// ## Why is this bad?
+/// `Path.with_suffix()` will raise an error at runtime
+/// if the given suffix is not prefixed with a dot.
+///
+/// ## Known problems
+/// This rule is prone to false positives and negatives
+/// due to type inference limitations, as it will only detect
+/// paths that are either instantiated (`p = Path(...)`)
+/// or annotated (`def f(p: Path)`) as such.
+///
+/// ## Fix safety
+/// The fix is marked as unsafe, since it changes runtime behaviour
+/// (`.with_suffix()` call no longer raises an exception).
+///
+/// ## Examples
+///
+/// ```python
+/// path.with_suffix("py")
+/// ```
+///
+/// Use instead:
+///
+/// ```python
+/// path.with_suffix(".py")
+/// ```
```

Comment:
Can you move this section above the `## Known problems` and `## Fix safety` section?